lib/bookstore-scrapper/indexTextbooks.mts

```diff
@@ -0,0 +1,51 @@
+/*
+    Run:
+    tsc --downlevelIteration --target es2021 --moduleResolution node --module es2022 --allowSyntheticDefaultImports indexTextbooks.mts; 
+    NODE_TLS_REJECT_UNAUTHORIZED='0' node indexTextbooks.mjs
+*/
+
+import { CampusStoreEntry } from './helpers/getTextbookInformation.mjs';
+import { updateDatabase } from './database/updateDatabase.mjs';
+import { getBookData } from './helpers/getBookData.mjs';
+import { axiosRequest } from './helpers/axiosRequest.mjs';
+import {
+    updateAlgoliaIndex,
+    getAlgoliaObject,
+} from './algolia/updateAlgoliaIndex.mjs';
+import fs from 'fs';
+
+// Algolia constants - FILL BEFORE RUNNING
+export const algoliaAppId: string = '';
+export const algoliaApiKey: string = '';
+export const algoliaIndexName: string = '';
+
+// Variables used to build links that connect to campus store
+const storeDomain = 'https://campusstore.mcmaster.ca';
+const storeBookPath = '/cgi-mcm/ws';
+const storeCourseMaterialPath = '/txhome.pl?wsgm=coursematerial';
+
+const indexTextbooks = async () => {
+    const courseMaterialHTML: string = await axiosRequest(
+        `${storeDomain}${storeBookPath}${storeCourseMaterialPath}`
+    );
+
+    const campusStoreData: CampusStoreEntry[] = await getBookData(
+        courseMaterialHTML
+    );
+
+    let data = JSON.stringify(campusStoreData);
+    fs.writeFileSync('data.json', '{ "Data":' + data + '}');
```

Comment:
Replacing this with the Promise-based version might result in a performance gain, since it won't block the main thread:
```typescript
import fs from "fs/promise";
...
const indexTextbooks = async () => {
    ...
    await fs.writeFile(...);
    ...
};
```